lms/djangoapps/grades/rest_api/v1/views.py

```diff
@@ -207,3 +219,318 @@ def _get_course(self, request, course_id):
+            )
+
+        if username:
+            username_filter = [Q(user__username=username)]
+        course_enrollments = self._paginate_course_enrollment(course_id, course_enrollment_filter=username_filter)
+
+        course_xblock_structure = SubmissionHistoryView._generate_course_structure(course_enrollments)
+        for course_key, course_info in course_xblock_structure.items():
+            course_data = SubmissionHistoryView._get_course_data(
+                course_key,
+                course_info.get('course_enrollments'),
+                course_info.get('course'),
+                course_info.get('blocks')
+            )
+            data.extend(course_data)
+        return self.get_paginated_response(data)
+
+    @staticmethod
+    def _generate_course_structure(enrollments):
+        """ Generate a map of course to course enrollment and problem
+            xblocks for each of the course.
+        """
+        course_enrollment_id_map = defaultdict(list)
+        course_xblock_structure = {}
+        for course_enrollment in enrollments:
+            course_enrollment_id_map[str(course_enrollment.course_id)].append(course_enrollment)
+        for course_key, course_enrollments in course_enrollment_id_map.items():
+            course_id = CourseKey.from_string(course_key)
+            course = get_course(course_id, depth=4)
+            course_xblock_structure[course_key] = {
+                'course_enrollments': course_enrollments,
+                'blocks': SubmissionHistoryView.get_problem_blocks(course),
+                'course': course
+            }
+        return course_xblock_structure
+
+    @staticmethod
+    def get_problem_blocks(course):
+        """ Get a list of problem xblock for the course.
+        """
+        blocks = []
+        for section in course.get_children():
+            for subsection in section.get_children():
+                for vertical in subsection.get_children():
+                    for block in vertical.get_children():
+                        if block.category == 'problem' and getattr(block, 'has_score', False):
+                            blocks.append(block)
+        return blocks
+
+    @staticmethod
+    def _get_course_data(course_key: str, course_enrollments: List[CourseEnrollment], course, blocks):
+        """
+        Get course data.
+
+        Params:
+        --------
+        course: course
+        block: XBlock to analyze.
+        """
+        course_grouped_data = []
+        for course_enrollment in course_enrollments:
+            course_data = {
+                'course_id': course_key,
+                'course_name': course.display_name_with_default,
+                'user': course_enrollment.user.username,
+                'problems': []
+            }
+            for block in blocks:
+                problem_data = SubmissionHistoryView._get_problem_data(course_enrollment, block)
+                course_data['problems'].append(problem_data)
```

Comment:
Should we introduce the following condition to filter out blocks that didn't get any answer from the user?

```suggestion
                if problem_data["submission_history"]:
                    course_data['problems'].append(problem_data)
```